runtime/src/bank.rs

```diff
@@ -4989,8 +4989,10 @@ impl Bank {
         #[cfg(not(test))]
         assert!(!validator_stakes.is_empty());
 
-        // Sort first by stake and then by validator identity pubkey for determinism
-        validator_stakes.sort_by(|(pubkey1, staked1), (pubkey2, staked2)| {
+        // Sort first by stake and then by validator identity pubkey for determinism.
+        // If two items are still equal, their relative order does not matter since
+        // both refer to the same validator.
+        validator_stakes.sort_unstable_by(|(pubkey1, staked1), (pubkey2, staked2)| {
```

Comment:
couldn't this be:
```rust
use std::cmp::Reverse;
validator_stakes.sort_unstable_by_key(|&(pubkey, stake)| Reverse((stake, pubkey)));
```

Reply:
If you want to avoid copying, then maybe:
```rust
validator_stakes.sort_by(|(pubkey1, staked1), (pubkey2, staked2)| {
    ((staked1, pubkey1)).cmp(&(staked2, pubkey2)).reverse()
});
```
would be more readable.